projects/igniteui-angular/src/lib/tabs/tabs.directive.ts

```diff
@@ -196,7 +196,7 @@ export abstract class IgxTabsDirective extends IgxCarouselComponentBase implemen
     protected onItemChanges() {
         this.setAttributes();
 
-        if (this.selectedIndex >= 0 && this.selectedIndex < this.items.length) {
+        if (this.selectedIndex < this.items.length) {

```

Comment:
Having a tab with `selected=true` should have higher priority than checking the old `selectedIndex`. I suggest to modify the method like this:
```ts
    protected onItemChanges() {
        this.setAttributes();

        // Check if there is selected tab
        let selectedIndex = -1;
        this.items.some((tab, i) => {
            if (tab.selected) {
                selectedIndex = i;
            }
            return tab.selected;
        });

        if (selectedIndex >= 0) {
            // Set the selected index to the tab that has selected=true
            Promise.resolve().then(() => {
                this.selectedIndex = selectedIndex;
            });
        } else {
            if (this.selectedIndex >= 0 && this.selectedIndex < this.items.length) {
                // Select the tab on the same index the previous selected tab was
                Promise.resolve().then(() => {
                    this.updateSelectedTabs(null);
                });
            } else if (this.selectedIndex >= this.items.length) {
                // Select the last tab
                Promise.resolve().then(() => {
                    this.selectedIndex = this.items.length - 1;
                });
            }
        }
    }
```